### PR TITLE
[IoT Hub] - update resource API versions & create the Bicep file

### PR DESCRIPTION
**Pull request by mumian**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@robinsh - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@robinsh - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@robinsh - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.devices/iothub-auto-route-messages/main.bicep [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      events: {
+        retentionTimeInDays: 1
+        partitionCount: d2cPartitions
+      }
+    }
+    routing: {
+      endpoints: {
+        storageContainers: [
+          {
+            connectionString: 'DefaultEndpointsProtocol=https;AccountName=${storageAccountName};EndpointSuffix=${environment().suffixes.storage};AccountKey=${listKeys(storageAccount.id, '2021-02-01').keys[0].value}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.devices/iothub-auto-route-messages/main.bicep:60]: remove listKeys and use ```storageAccount.listKeys.keys[0].value```

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@robinsh - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**